server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -129,6 +135,73 @@ export class AssetService {
     }
   }
 
+  public async importFile(authUser: AuthUserDto, dto: ImportAssetDto): Promise<AssetFileUploadResponseDto> {
+    dto = {
+      ...dto,
+      assetPath: path.resolve(dto.assetPath),
+      sidecarPath: dto.sidecarPath ? path.resolve(dto.sidecarPath) : undefined,
+    };
+
+    const assetPathType = mime.lookup(dto.assetPath) as string;
+    if (!isSupportedFileType(assetPathType)) {
+      throw new BadRequestException(`Unsupported file type ${assetPathType}`);
+    }
+
+    if (dto.sidecarPath) {
+      const sidecarType = mime.lookup(dto.sidecarPath) as string;
+      if (!sidecarType.match(/\/xml$/)) {
```

Comment:
```suggestion
      if (!['application/xml', 'text/xml'].includes(file.mimetype)) {
```

It might be better to make a function for this like was done for `isSupportedFileType`.